### PR TITLE
Anthony McKeever - Lesson 2 Exercises 1, 2, and 3 Submission

### PR DESCRIPTION
**Pull request by Snip3rM00n**
Implements the following course materials:
Lesson 2 - Exercise 1 - Grid Printer
This was a fun exercise!  I had a nice challenge and ended up really considering ways to improve my overall code quality.  While I was doing this exercise my internet went out and I was unable to continue the course for the night so I ended up taking time to optimize the code.  I hope I didn't go overboard.  `^^;;`

Lesson 2 - Exercise 2 - Fizz Buzz
Typical Fizz Buzz!  Always good to go back to basics.

Lesson 2 - Exercise 3 - Fibonacci (and Lucus) Series
Had to research Lucas numbers, I had never heard of them before.
(Incidentally, Fibonacci and Lucas would make a great name for a sitcom)

---

Squashed Commit History:

* Lesson 2 Exercise 1: Grid Printer

Implement a method of printing grids to the console.

I may have gone overboard on this with the refactoring and optimization.
I had nothing else better to do when the internet went out at home.  I
hope it still keeps in with the spirit of the exercise.

* Adds Lesson 2 Exercise 2 - FizzBuzz

Updates ReadMe to include additional folder structure.
Updates "task" in grid_printer to "Exercise"
Adds FizzBuzz assignment work.
	Adds two scoops of Fizz in every Buzz

* Add Series Exercise

Implement Fibonacci Calculations
Implement Lucas Calculations
Implement General Function (can do both fib and lucas)
Add Unit Tests

* Folders and ReadMe

Update ReadMe with new folder structure
Add ending blank line in ReadMe	(appease the GitHub Gods
Rename folders from exercise names to exercise numbers.

* Fix Readme Markdown

* Optimize printMe string to eliminate for loop

Removes print each char for loop and replaces it with joining the string
together with spaces between each character.

Correct exercise name

### CODE REVIEW

**Comment by Snip3rM00n** [comment]
A couple questions for the professors:

1. Do we, as a class, prefer function and symbol names with underscores over camel case?  For example is `my_function()` preferred over `myFunction()`?

1. I've been using a branching strategy in my personal repo and using a pull request to help me review each assignment to ensure it meets the assignment requirements and the code is good quality.  When I merge the branches into my master before submitting a pull request of my master to the class master I've been squashing my commits.  Is this okay or would ya'll prefer that I not squash?

**Comment by natasha-aleksandrova** [comment]
@Snip3rM00n we follow pep8 style guide, in which case you would use the snake_case convention. If you use something like Pycharm, it will follow these same conventions.

As far as branching, I would only recommend for someone comfortable with github to do so. We want to avoid any git issues so that you concentrate on the coding part. As far as squashing commits, no preference here.

File: students/anthony_mckeever/lesson2/exercise_3/series.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    :param n:   The number of the place in the Lucas series to get.  For example, n = 7 will return 29
+    """
+    if n == 0:
+        return 2
+    if n == 1:
+        return 1
+
+    return lucas(n - 1) + lucas(n - 2)
+
+def sum_series(n, atZero = 0, atOne = 1):
```
Comment (charlesrobison) [inline students/anthony_mckeever/lesson2/exercise_3/series.py]: Make these camel_case too (almost everything is camel_case in Python except Class names :) . Also, no whitespace around the equal signs.

File: students/anthony_mckeever/lesson2/exercise_3/series.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    for i in range(8):
+        assert sum_series(i, 2) == lucas(i)
+        assert sum_series(i, 2, 1) == lucas(i)
+
+    print("Validate arbitrary values in sum_series")
+    test_expectations = [3, 2, 5, 7, 12, 19]
+    for i in range(len(test_expectations)):
+        assert sum_series(i, 3, 2) == test_expectations[i]
+
+    print("Tests passed!")
```
Comment (charlesrobison) [inline students/anthony_mckeever/lesson2/exercise_3/series.py]: Nice job on this!

File: students/anthony_mckeever/lesson2/exercise_1/grid_printer.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    """
+    Print the template to the console to form the grid
+
+    :param template:    The Horizontal or Vertical template to print on this call.
+    :param multiple:    How many times to repeat the template in the current line.
+    :param endChar:     The ending character of the line.
+    :param times:       How many times to print the same template in the console
+                        on a new line.
+    """
+    printMe = " ".join((template * multiple) + endChar)
```
[the remark this answers is not in the document]
Comment (Snip3rM00n) [reply students/anthony_mckeever/lesson2/exercise_1/grid_printer.py]: C# habits die hard.  I will do my level best to switch to `snake_case` from `camelCase` for future course work.  🙂  Should I fix for this PR or just keep it in mind going forward?

File: students/anthony_mckeever/lesson2/exercise_3/series.py (hunk 2 above)
Comment (Snip3rM00n) [reply students/anthony_mckeever/lesson2/exercise_3/series.py]: Thank you!  I have a passion for comprehensive and verbose testing.  😁

File: students/anthony_mckeever/lesson2/exercise_1/grid_printer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (charlesrobison) [reply students/anthony_mckeever/lesson2/exercise_1/grid_printer.py]: You received full credit, so no need to fix unless you'd like to.  :)

**Pull request merged**